### PR TITLE
feat: add unsubscribe exmaple for go

### PR DESCRIPTION
**Pull request by xavier-hou**
添加一个函数来示例取消订阅

### CODE REVIEW

**Comment by xavier-hou** [comment]
@zibuyu1995  PTAL

File: mqtt-client-Go/on_way-tls/main.go [hunk 1]
```diff
@@ -39,37 +40,50 @@ func main() {
 		Cafile:   *cafile,
 	}
 	client := mqttConnect(&config)
-	go sub(client, &config)
+	go subscribe(client, &config)
```
Comment (zibuyu1995) [inline mqtt-client-Go/on_way-tls/main.go:43]: You can add the subscription to the OnConnect callback.

File: mqtt-client-Go/on_way-tls/main.go [hunk 2]
```diff
@@ -39,37 +40,50 @@ func main() {
...
 	publish(client, &config)
+	time.Sleep(time.Second)
+	unSubscribe(client, &config)
+	time.Sleep(time.Second)
+	client.Disconnect(250)
 }
 
 func publish(client mqtt.Client, config *ConConfig) {
-	for {
```
Comment (zibuyu1995) [inline mqtt-client-Go/on_way-tls/main.go:47]: You can keep the loop statement and just add a limit on the number of iterations.

**Review by zibuyu1995 — CHANGES_REQUESTED** [review]

File: mqtt-client-Go/on_way-tls/main.go (hunk 2 above)
Comment (xavier-hou) [reply mqtt-client-Go/on_way-tls/main.go:47]: done

File: mqtt-client-Go/on_way-tls/main.go (hunk 1 above)
Comment (xavier-hou) [reply mqtt-client-Go/on_way-tls/main.go]: done

**Comment by xavier-hou** [comment]
@zibuyu1995  执行日志
```
2023/05/10 13:22:05 Connected to MQTT broker
2023/05/10 13:22:05 pub [emqx/golang-mqtt] The current time 2023-05-10 13:22:05.109883 +0300 IDT m=+0.594463835
2023/05/10 13:22:05 Subscribe to topic:emqx/golang-mqtt successfully
2023/05/10 13:22:06 pub [emqx/golang-mqtt] The current time 2023-05-10 13:22:06.111539 +0300 IDT m=+1.596108168
2023/05/10 13:22:06 sub [emqx/golang-mqtt] The current time 2023-05-10 13:22:06.111539 +0300 IDT m=+1.596108168
2023/05/10 13:22:07 pub [emqx/golang-mqtt] The current time 2023-05-10 13:22:07.112749 +0300 IDT m=+2.597307210
2023/05/10 13:22:07 sub [emqx/golang-mqtt] The current time 2023-05-10 13:22:07.112749 +0300 IDT m=+2.597307210
2023/05/10 13:22:08 pub [emqx/golang-mqtt] The current time 2023-05-10 13:22:08.113249 +0300 IDT m=+3.597795085
2023/05/10 13:22:08 Unsubscribe from topic:emqx/golang-mqtt successfully
2023/05/10 13:22:09 pub [emqx/golang-mqtt] The current time 2023-05-10 13:22:09.114377 +0300 IDT m=+4.598911668
2023/05/10 13:22:10 pub [emqx/golang-mqtt] The current time 2023-05-10 13:22:10.115074 +0300 IDT m=+5.599596668
2023/05/10 13:22:11 pub [emqx/golang-mqtt] The current time 2023-05-10 13:22:11.116315 +0300 IDT m=+6.600825793
2023/05/10 13:22:11 Disconnected from MQTT broker
```

**Review by Zhiw — APPROVED** [review]

**Review by wuzufeng — APPROVED** [review]

**Review by zibuyu1995 — APPROVED** [review]

**Pull request merged**